### PR TITLE
lib/scrape.js: Add year variable

### LINKED ISSUE #153: Add year variable, set to 2018
https://codein.withgoogle.com/api/program/2017/organization/?status=2 is 404.

https://codein.withgoogle.com/api/program/2018/organization/?status=2 is at least valid json

Ideally the year variable is set to the previous or current year, *depending* on which JSON of the above is the best.

i.e. on Jan 1 2019, the website still needs to be using 2018 , and should keep using 2018 until the 2019 org data is more valid.

### PR DESCRIPTION
**Pull request by utkarsh2102**
This fixes the year by creating a GCI_YEAR variable.

Fixes https://github.com/coala/gci-leaders/issues/153

### CODE REVIEW

**Review by pandyamarut — APPROVED** [review]
Hopefully this changes works. 
@jayvdb Look into it.

File: lib/scrape.js [hunk 1]
```diff
@@ -97,12 +98,13 @@ try {
...
 async function fetchProgram() {
-  const res = await fetch(`${GCI_API_BASE}/program/2017/`)
+  const res = await fetch(`${GCI_API_BASE}/program/${GCI_YEAR}/`)
   return await res.json()
 }
 
 async function fetchOrgs() {
-  const res = await fetch(`${GCI_API_BASE}/program/2017/organization/?status=2`)
+  const res = await fetch(
+        `${GCI_API_BASE}/program/${GCI_YEAR}/organization/?status=2`)
```
Comment (li-boxuan) [inline lib/scrape.js]: Please make CI happy. See https://travis-ci.org/coala/gci-leaders/builds/438548213#L1059

**Review by li-boxuan — CHANGES_REQUESTED** [review]

File: lib/scrape.js [hunk 2]
```diff
@@ -17,6 +17,7 @@ const GH_ORG_BASE = `${GH_BASE}/orgs`
 const GH_API_BASE = 'https://api.github.com'
 const GH_GQL_BASE = 'https://api.github.com/graphql'
 const GCI_API_BASE = 'https://codein.withgoogle.com/api'
+const GCI_YEAR = '2018'
```
Comment (jayvdb) [inline lib/scrape.js]: the year needs to be dynamic.

**Review by jayvdb — CHANGES_REQUESTED** [review]
Please squash your commits.
We only want one commit per atomic change.

**Comment by utkarsh2102** [comment]
Sure @jayvdb.
I'll use the `(new Date()).getFullYear()` function to make it dynamic.

File: lib/scrape.js [hunk 3]
```diff
@@ -97,12 +98,14 @@ try {
...
 async function fetchProgram() {
-  const res = await fetch(`${GCI_API_BASE}/program/2017/`)
+  const res = await fetch(`${GCI_API_BASE}/program/${GCI_YEAR}/`)
   return await res.json()
 }
 
 async function fetchOrgs() {
-  const res = await fetch(`${GCI_API_BASE}/program/2017/organization/?status=2`)
+  const res = await fetch(
+    `${GCI_API_BASE}/program/${GCI_YEAR}/organization/?status=2`
```
Comment (li-boxuan) [inline lib/scrape.js]: Please add logic: If this address is invalid, then try the previous year.
As https://github.com/coala/gci-leaders/issues/153#issue-358459483 says, on Jan 1 2019, the website still needs to be using 2018, and should keep using 2018 until the 2019 org data is more valid.

**Review by li-boxuan — CHANGES_REQUESTED** [review]
Your commit message has a typo.

File: lib/scrape.js (hunk 3 above)
Comment (utkarsh2102) [reply lib/scrape.js]: Oh, okay.
Working on it.

**Comment by jayvdb** [comment]
fwiw, the netlify build is failing in `node ./lib/scrape.js` with `4:49:43 AM: (node:1600) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): TypeError: Cannot read property 'repositories' of null` . very likely this is an unrelated problem, but we'll need to get it fixed too.

**Comment by utkarsh2102** [comment]
Yes, @jayvdb.
Moreover, I think @li-boxuan pulled my changes and made a PR(https://github.com/coala/gci-leaders/pull/161), too.
I think that'd resolve the problem.

**Review by jayvdb — CHANGES_REQUESTED** [review]
The year on the front page is still 2017.

**Review by li-boxuan — APPROVED** [review]
LGTM ;)
We still have hard-coded year number in i18n files, but we can tackle that later.

File: lib/scrape.js [hunk 4]
```diff
@@ -95,15 +95,31 @@ try {
...
+let year = new Date().getFullYear()
 async function fetchProgram() {
-  const res = await fetch(`${GCI_API_BASE}/program/2017/`)
-  return await res.json()
+  const res = await fetch(`${GCI_API_BASE}/program/${year}/`)
+  if (res.status == 200) {
+    return await res.json()
+  }
+
+  year -= 1
```
Comment (li-boxuan) [inline lib/scrape.js:105]: This looks ugly :P 
```javascript
const year = new Date().getFullYear()
const lastYear = year - 1
```
is slightly better IMO

Comment (li-boxuan) [reply lib/scrape.js:105]: Nvm it is working anyway, if there is improvement we can do later

**Pull request merged**